### PR TITLE
[CEN-722] Secrets and Configmaps for FA Customer Microservice

### PR DESCRIPTION
**Pull request by GiovanniMancini**
This PR proposes to add standard secrets and configmap to start FA Customer microservice.

### List of changes

- A new secret to connect to Postgres
- A new  secret to send logs to App Insight
- A basic configmap for the microservice

### Motivation and context

The deployment yaml references secrets and configmaps which don't exist.

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/k8s/fa_configmaps.tf [hunk 1]
```diff
@@ -21,3 +21,17 @@ resource "kubernetes_config_map" "fa-jvm" {
     TZ = "Europe/Rome"
   }
 }
+
+
+resource "kubernetes_config_map" "fa-ms-customer" {
```
Comment (pasqualedevita) [inline src/k8s/fa_configmaps.tf:26]:
```suggestion
resource "kubernetes_config_map" "famscustomer" {
```
only for take a standard naming convention

File: src/k8s/fa_configmaps.tf [hunk 2]
```diff
@@ -21,3 +21,17 @@ resource "kubernetes_config_map" "fa-jvm" {
...
+  metadata {
+    name      = "famscustomer"
+    namespace = kubernetes_namespace.fa.metadata[0].name
+  }
+
+  data = merge({
+    POSTGRES_SCHEMA = "fa_customer"
+    TZ = "Europe/Rome"
+  }, var.configmaps_facustomer_override)
```
Comment (pasqualedevita) [inline src/k8s/fa_configmaps.tf:35]:
```suggestion
  }, var.configmaps_facustomer)
```
only for take a standard naming convention

File: src/k8s/subscriptions/DEV-CSTAR/terraform.tfvars [hunk 3]
```diff
@@ -210,3 +210,13 @@ configmaps_rtdpaymentinstrumentmanager = {
   POSTGRES_SHOW_SQL                                 = "true"
   LOG_LEVEL_RTD_PAYMENT_INSTRUMENT_MANAGER          = "INFO"
 }
+
+
+configmaps_facustomer_override = {
```
Comment (pasqualedevita) [inline src/k8s/subscriptions/DEV-CSTAR/terraform.tfvars:215]:
```suggestion
configmaps_facustomer = {
```
for naming convention

File: src/k8s/variables.tf [hunk 4]
```diff
@@ -154,4 +154,9 @@ variable "configmaps_bpdmswinningtransaction" {
 
 variable "configmaps_rtdpaymentinstrumentmanager" {
   type = map(string)
+}
+
+
+variable "configmaps_facustomer_override" {
```
Comment (pasqualedevita) [inline src/k8s/variables.tf:160]:
```suggestion
variable "configmaps_facustomer" {
```
for naming convention

File: src/k8s/fa_configmaps.tf [hunk 5]
```diff
@@ -21,3 +21,17 @@ resource "kubernetes_config_map" "fa-jvm" {
...
+
+resource "kubernetes_config_map" "fa-ms-customer" {
+  metadata {
+    name      = "famscustomer"
+    namespace = kubernetes_namespace.fa.metadata[0].name
+  }
+
+  data = merge({
+    POSTGRES_SCHEMA = "fa_customer"
+    TZ = "Europe/Rome"
```
Comment (pasqualedevita) [inline src/k8s/fa_configmaps.tf:34]: in others `bpd`/`rtd` ms we used a configmap ref jvm (https://github.com/pagopa/bpd-ms-winning-transaction/blob/release-prod/manifests/pagoPa/deployment-prod.yml#L37) that is missing for `fa-ms-customer `https://github.com/pagopa/fa-ms-customer/blob/release-dev/manifests/pagoPa/deployment-dev.yml)

In my opinion, if there isn't a new requiriment/improvement it's better to use a standard model

Also can we add all possibile variables with the default value? in this way on main we have the list of all possibile variables and their values for a quick and dirty deploy

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: src/k8s/fa_configmaps.tf (hunk 1 above)
Comment (GiovanniMancini) [reply src/k8s/fa_configmaps.tf:26]: yes, I agree. thank you

File: src/k8s/fa_configmaps.tf (hunk 5 above)
Comment (GiovanniMancini) [reply src/k8s/fa_configmaps.tf:34]: @pasqualedevita Could I consider this comment resolved and try to apply the PR in dev?

**Review by pasqualedevita — APPROVED** [review]

**Pull request closed without merging**